### PR TITLE
adds font-size conversion logic for scaling

### PR DESCRIPTION
**Pull request by esotrope**

### CODE REVIEW

File: src/main/java/org/opentestsystem/ap/item/translation/mappers/qti/content/FontSizedElementProcessor.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+            final String[] keyAndValue = rule.split(":");
+            if (keyAndValue.length == 2) {
+                rules.add(Pair.of(
+                    keyAndValue[0].trim(),
+                    keyAndValue[1].trim()
+                ));
+            }
+        }
+        return rules;
+    }
```
Comment (esotrope) [inline src/main/java/org/opentestsystem/ap/item/translation/mappers/qti/content/FontSizedElementProcessor.java]: if we need to do other style/css processing this and the other methods here could be extracted

File: src/main/java/org/opentestsystem/ap/item/translation/mappers/qti/content/FontSizedElementProcessor.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    private String pixelsToEphemerals(final String value, final double baseFontSize) {
+
+        return replaceAll(
+            value,
+            pixelSizePattern,
+            matcher -> {
+                final double size = Double.parseDouble(matcher.group(1));
+                final double relativeSize = size / baseFontSize;
+                return String.valueOf(relativeSize)
+                    .replaceAll("\\.0+$", "")
```
Comment (esotrope) [inline src/main/java/org/opentestsystem/ap/item/translation/mappers/qti/content/FontSizedElementProcessor.java]: removes .0000 (i guess there could be non-zeroes in between the period and the trailing zeros hm.)

File: src/main/java/org/opentestsystem/ap/item/translation/mappers/qti/content/FontSizedElementProcessor.java [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+    /**
+     * Finds and replaces all matches given the provided pattern and replacer callback
+     *
+     * @param value The value to match
+     * @param pattern The pattern to match with
+     * @param replacer Function accepting the match and returning the replacement value
+     * @return The new string with replaced values
+     */
+    private String replaceAll(final String value, final Pattern pattern, final Function<Matcher, String> replacer) {
```
Comment (esotrope) [inline src/main/java/org/opentestsystem/ap/item/translation/mappers/qti/content/FontSizedElementProcessor.java]: i was surprised to find this was not in the jdk or string utils

File: src/main/java/org/opentestsystem/ap/item/translation/mappers/qti/content/FontSizedElementProcessor.java (hunk 1 above)
Comment (wtritch) [reply src/main/java/org/opentestsystem/ap/item/translation/mappers/qti/content/FontSizedElementProcessor.java]: I'd almost be tempted to use a CSS parsing lib, if it gets more complex.

File: src/main/java/org/opentestsystem/ap/item/translation/mappers/qti/content/FontSizedElementProcessor.java (hunk 2 above)
Comment (wtritch) [reply src/main/java/org/opentestsystem/ap/item/translation/mappers/qti/content/FontSizedElementProcessor.java]: Maybe switch the pattern to "[^\\.]0+$" which should remove all trailing zeros *except* the one immediately following a period?

Comment (esotrope) [reply src/main/java/org/opentestsystem/ap/item/translation/mappers/qti/content/FontSizedElementProcessor.java]: yeah - actually almost feel like just removing this as it is not required and just can make the resulting css look cleaner

File: src/main/java/org/opentestsystem/ap/item/translation/mappers/qti/content/FontSizedElementProcessor.java (hunk 3 above)
Comment (wtritch) [reply src/main/java/org/opentestsystem/ap/item/translation/mappers/qti/content/FontSizedElementProcessor.java]: Yeah, you can have simple group replacement, but no processing replacement by default that I'm aware of.

**Pull request merged**